Performs various cleanups and updates to adjust archetypes

lukaszlenart opened the pull request.

lukaszlenart committed:
Performs various cleanups and updates to adjust archetypes

lukaszlenart committed:
Uses more compact description which is used when generating artifacts

JCgH4164838Gh792C124B5 commented:
Hi @lukaszlenart .

The cleanup changes look good to me.  👍 

The Struts Dev list mentioned a couple of possible cleanup items for the archetypes that were noted in the project created by a given archetype: "
1) `struts2-archetype-portlet` - The pom.xml generated by the archetype uses: "<struts2.version>${supported.struts2.version}</struts2.version>" but "`supported.struts2.version`" does not appear to be defined anywhere in the POM hierarchy by default and caused a build failure for the module. Replacing "${supported.struts2.version}" with a **literal** "2.5.20" built fine.

2) `struts2-archetype-plugin` - Since no explicit source/target exist in the pom.xml Maven ends up using "`-target 1.5 -source 1.5`" in the build. It builds fine with JDK 8, but fails when building with JDK 11.  Forcing a maven-compiler-plugin entry in pom.xml with `source/target 1.7` fixed the issue for JDK 11.".

I think both would involve a changes to the pom.xml in "`archetype-resources`" for each archetype (properly defining "supported.struts2.version" in the first, adding a "maven-compiler-plugin" entry for source/target 1.7 in the second".

Since this PR is aimed at cleanup, maybe those two items above might be something to add to it ?

If you think they should be handled in a separate PR I could try it out - let me know what you prefer.

lukaszlenart committed:
Uses Maven Archetype plugin 3.1.3 which supports skipping non-archetypes
This functionality is under review https://github.com/apache/maven-archetype/pull/33

lukaszlenart committed:
Drops empty line

lukaszlenart commented:
I'm waiting on this PR https://github.com/apache/maven-archetype/pull/33 to move forward

lukaszlenart committed:
Adds repos with SNAPSHOT plugins

lukaszlenart commented:
The PR got merged, these changes can be reviewed now.

yasserzamani commented:
Sorry I don't have enough experience with archetypes to review now but in general as first step I think this project should have some basic general tests e.g. please see ["I've created a few archetypes for a project that work fine for now, but I'd like to be able to verify that the code generated from the archetypes continues to work in the future"](https://stackoverflow.com/q/4082643/1362623).

JCgH4164838Gh792C124B5 commented on pom.xml:
```diff
@@ -107,6 +100,36 @@
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-archetype-plugin</artifactId>
+                    <version>3.1.3-SNAPSHOT</version>
```
Having a `3.1.3-SNAPSHOT` version of the maven-archetype-plugin referenced here caused my local builds of the cleanup branch code to fail.  I'm assuming that it will create the same issue for most users too.

Changing version to `3.1.2` (latest full release) allowed the archetype project to build successfully (JDK 11).  That is the version of `archetype-packaging` being referred to in other changes for this PR (if that makes a difference).

I _think_ it should be set to `3.1.2` for this PR before it is merged ?

lukaszlenart replied:
Would you mind double checking this again? I fixed the SNAPSHOT repository definition to allow fetch a proper version of the plugin.

JCgH4164838Gh792C124B5 replied:
The pom change to use `pluginRepositories` seemed to fix the issue for me (allowed proper download of the 3.1.3-SNAPSHOT).  :smile:

If that snapshot is preferred over 3.1.2 final (for whatever reason) the build does work now.

lukaszlenart replied:
SNAPSHOTs are fine but once we decide to prepare a release we must turn all SNAPSHOTs into proper versions. Hopefully they will release a new version till then :)

JCgH4164838Gh792C124B5 commented on pom.xml:
```diff
@@ -107,6 +100,36 @@
+        </pluginManagement>
     </build>
 
+    <!-- Remove once maven-archetype-plugin get released-->
```
Even with this section present I had issues building the archetypes project (within NetBeans 11.1).  Maybe it was a transient issue but it might be better to stick with 3.1.2 as it seems to build fine with that version anyway ?

JCgH4164838Gh792C124B5 reviewed (changes requested):
Hello @lukaszlenart.

The PR LGTM with the possible exception of the `maven-archetype-plugin` version being a snapshot.  There is also a really minor fix for JDK 11+ that was found when reviewing this PR (it is in PR #4 and could be added to this PR directly if you want).

The cleanup branch code built successfully using JDK 11 with NetBeans 11.1.  Using the struts-archetypes local snapshot a test project with one module for each archetype was generated and they all compiled successfully (except for the `struts2-archetype-plugin` result which needs the minor fix in #4 to compile with JDK 11).

This is my first direct experience with maven archetypes so something may have been missed in the review.

Otherwise the changes LGTM. :clap:

lukaszlenart committed:
Uses pluginRepository to allow fetch SNAPSHOT plugins

JCgH4164838Gh792C124B5 reviewed (approved):
LGTM. 👍 
The latest changes work for me locally.  If anyone sees any issues they can advise.
